services/github/github-release.service.js

```diff
@@ -39,22 +92,106 @@ module.exports = class GithubRelease extends GithubAuthService {
     }
   }
 
-  static render({ version, isPrerelease }) {
-    return {
-      message: addv(version),
-      color: isPrerelease ? 'orange' : 'blue',
+  static render({ version, sort, isPrerelease }) {
+    let color = 'blue'
+    color = sort === 'semver' ? versionColor(version) : color
+    color = isPrerelease ? 'orange' : color
+    return { label: 'release', message: addv(version), color }
+  }
+
+  async fetchReleases({ user, repo }) {
+    return this._requestJson({
+      url: `/repos/${user}/${repo}/releases`,
+      schema: releaseInfoArraySchema,
+      errorMessages: errorMessagesFor('no releases or repo not found'),
+    })
+  }
+
+  static getLatestRelease({ releases, sort, includePrereleases }) {
+    if (sort === 'semver') {
+      const latestRelease = latest(releases.map(release => release.tag_name), {
+        pre: includePrereleases,
+      })
+      const kvpairs = Object.assign(
+        ...releases.map(release => ({ [release.tag_name]: release.prerelease }))
+      )
+      return { tag_name: latestRelease, prerelease: kvpairs[latestRelease] }
+    }
+    if (!includePrereleases) {
+      const stableReleases = releases.filter(release => !release.prerelease)
+      if (stableReleases.length > 0) {
+        return stableReleases[0]
+      }
+      return releases[0]
     }
+    return releases[0]
   }
 
-  async handle({ variant, user, repo }) {
-    const {
-      tag_name: version,
-      prerelease: isPrerelease,
-    } = await fetchLatestRelease(this, {
-      user,
-      repo,
-      includePre: variant === 'release-pre',
+  async handle({ user, repo }, queryParams) {
+    const sort = queryParams.sort
+    const includePrereleases = queryParams.include_prereleases !== undefined
+
+    let latestRelease
+    if (!includePrereleases && sort === 'date') {
+      latestRelease = await fetchLatestRelease(this, { user, repo })
+      return this.constructor.render({
+        version: latestRelease.tag_name,
+        sort,
+        isPrerelease: latestRelease.prerelease,
+      })
+    }
+
+    const releases = await this.fetchReleases({ user, repo })
+    if (releases.length === 0) throw new NotFound({ prettyMessage: 'none' })
```

Comment:
Minor error message suggestion:
```suggestion
    if (releases.length === 0) throw new NotFound({ prettyMessage: 'no releases found' })
```

Reply:
Also, if we moved this line into the `getLatestRelease` function then I think this scenario would be easier to test via a unit test in the spec file.